plrustc/plrustc/uitests/static_impl_unsound.rs

```diff
@@ -0,0 +1,27 @@
+#![crate_type = "lib"]
+
+use std::fmt::Display;
+
+trait Displayable {
+    fn display(self) -> Box<dyn Display>;
+}
+
+// This is more complex than the one in the issue, to make sure the `Box`'s
+// lang_item status doesn't bite us.
+impl<T: Display> Displayable for (T, Box<Option<&'static T>>) {
```

Comment:
This is making me wonder: `Box<T>` is a lang item, but more than that, it is `#[fundamental]`. Should we have any special concerns about other `#[fundamental]` types here?

Reply:
Not as far as I know.

Reply:
To be more concrete, we are looking somewhat blindly for `'static` through almost everything, so fundamental or not, it doesn't make a huge difference.